src/python/pants/backend/project_info/tasks/export_dep_as_jar.py

```diff
@@ -326,15 +310,7 @@ def _dependencies_needed_in_classpath(target):
             if hasattr(current_target, "runtime_platform"):
                 info["runtime_platform"] = current_target.runtime_platform.name
 
-        transitive_targets = OrderedSet(
-            [
-                dep.address.spec
-                for dep in dependencies_needed_in_classpath
-                if dep in modulizable_target_set
-            ]
-        )
-        transitive_targets.update(info["targets"])
-        info["source_dependencies_in_classpath"] = [dep for dep in transitive_targets]
+        info["source_dependencies_in_classpath"] = self._compute_transitive_source_dependencies(current_target, info["targets"], modulizable_target_set)
```

Comment:
Hm, so it looks like order will matter. Refer to the above comment about how it is not safe to use `set` (unless you sort after the matter).